### PR TITLE
feat(moe): EP experts forward收纳到kernel底下实现多impls管理，并且添加 NPU grouped matmul的forward分支。

### PR DESCRIPTION
**Pull request by werwrewe**
kernelize 的 NPU patch 无法生效，导致NPU forward被ep forward覆盖。考虑在ep forward中单独给NPU提供forward。ep forward原实现逐 expert Python 循环，在 EP 闭包内部复用现有 NPU 组件（GmmFunction + npu_swiglu + _get_cached_expert_weights）替换循环实现并行化。每次前向做资格检查，不满足条件自动回退原循环并打印一次 WARNING 说明原因。

端到端验证： loss 曲线与原实现一致，使用ep_fsdp2_lora_qwen3_5_moe.sh 设备2*910B 模型qwen3.5-35B-A3B 每 12 step 约 200s -> 60s。

二编：
改动代码结构，原有EP experts forward也收纳到kernel底下ops里面并且实现多impls管理。测试两种ep forward分支（NPU和逐expert循环）都能跑通，使用ep_fsdp2_lora_qwen3_5_moe.sh 设备2*910B batch8 模型qwen3.5-35B-A3B 16steps NPU时间185s，逐expert循环393s。

### CODE REVIEW

File: src/twinkle/model/transformers/moe/ep_utils.py [hunk 1]
```diff
@@ -203,6 +207,13 @@ def preprocess(
     )
     dist.all_gather_into_tensor(num_global_tokens_per_expert, num_local_tokens_per_expert, group=ep_group)
 
+    # On NPU, the HCCL collective runs on a separate stream and its completion
+    # is not always ordered with the host reads (.tolist()/.item()) below —
+    # observed in practice as garbage split sizes. Force a device sync before
+    # reading gathered results back on the host.
+    if num_local_tokens_per_expert.device.type == 'npu':
+        torch.npu.synchronize()
```
Comment (kevssim) [inline src/twinkle/model/transformers/moe/ep_utils.py]: 最好使用平台无关的封装：
  from twinkle import torch_util

  torch_util.synchronize()

File: src/twinkle/model/transformers/moe/expert_parallel.py [hunk 2]
```diff
@@ -342,6 +442,14 @@ def ep_forward(
             # all-to-all, causing EP collective order divergence.
             return permuted_tokens
 
+        if _ep_experts_can_use_npu_gmm(self):
```
Comment (kevssim) [inline src/twinkle/model/transformers/moe/expert_parallel.py]: 更好的做法是把gmm定义成一个接口，有不同的impl，然后在expert_parallel.py中直接调用接口，隐藏impl细节。

**Review by kevssim — APPROVED** [review]
